### ISSUE TITLE
`winget upgrade` finds update but does nothing

### ISSUE THREAD

**Issue by sba923** [microsoft/winget-cli#3280]
### Brief description of your issue

`winget upgrade` tells me that at least one update is available, but does not install it:

![image](https://github.com/microsoft/winget-cli/assets/12860484/42bd3135-7bb6-4734-ae53-3ddf6971dfa2)

### Steps to reproduce

Run `winget upgrade` where `winget list` reports at least one entry in the `Available` column.

### Expected behavior

The corresponding application should get updated.

### Actual behavior

No update takes place.

### Environment

```shell
Windows Package Manager (Preview) v1.5.101-preview
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.1778
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.20.101.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

User Settings: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by mdanish-kh** [comment]
`winget list` lists down all apps installed on your machine and tells you whether those apps are available via any configured source (WinGet Packages Repository and Microsoft Store are the configured sources by default). `winget upgrade` by itself is basically a filter for `winget list` to show packages that have an upgrade available through any of the configured sources. You can think of `winget upgrade` as just `winget list --upgrade-available` (which is a valid command that was added in the latest preview version of WinGet)

To upgrade, you either want to run `winget upgrade --all`, or use the package ID `winget upgrade Microsoft.VisualStudio.2022.Community`

**Comment by sba923** [comment]
Oops, that's exactly what I'm doing... Will edit the issue...

**Comment by mdanish-kh** [comment]
Okay so I could also reproduce the problem. It's due to the fact that the installation is machine-wide but `Scope: user` is used in VS manifests due to which WinGet is not able to upgrade the package. I can update the manifest but it will probably require a review from VS Team

**Comment by sba923** [comment]
Is this related to a recent change? This used to work fine....

**Comment by mdanish-kh** [comment]
Yes, this is a recent change. The older manifests didn't use to contain the optional `Scope` value

**Comment by sba923** [comment]
IMVHO two things should be done about this issue:
1. fix the manifest for the specific VS2022 case
2. teach `winget` to display a message explaining why no upgrade is / can be done in such a situation
